src/net.cpp

```diff
@@ -345,9 +345,6 @@ bool CConnman::CheckIncomingNonce(uint64_t nonce)
 CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCountFailure)
 {
     if (pszDest == NULL) {
-        if (IsLocal(addrConnect))
```

Comment:
This changes behavior for "normal" nodes too, should probably avoid that. I think passing `fConnectToMasternode` to `CConnman::ConnectNode` and using it here would be a better solution.